design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,254 @@
+# Spike - Containerless CLI
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+  * [Useful links](#useful-links)
+  * [Background](#background)
+      - [Requirements](#requirements)
+  * [Research](#research)
+      - [Estimation](#estimation)
+      - [Takeways and recommendation](#takeways-and-recommendation)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+| **Term**                                      | **Description**                                              |
+| --------------------------------------------- | ------------------------------------------------------------ |
+| Epic                                          | https://github.com/cyberark/conjur/issues/1892               |
+| Ruby CLI                                      | https://github.com/cyberark/conjur-cli                       |
+| Go CLI                                        | https://github.com/cyberark/conjur-cli-go                    |
+| Python CLI                                    | https://github.com/cyberark/conjur-api-python3               |
+| [FPM](https://github.com/jordansissel/fpm)    | For building packages for multiple platforms (RPM, DEB, OS X) |
+| [Debify](https://github.com/conjurinc/debify) | Build a package from a Ruby gem                              |
+
+## Background
+
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+
+This document is one piece in the bigger piece of the Dockerless Conjur puzzle. We need to be able to deploy Conjur on a VM. This means we will need to figure out a way to deliver a containerless CLI to allow human-users to interface with Conjur.
+
+In this document we will discuss possible options for a containerless CLI and present a recommendation for which one we should pursue.
+
+#### Requirements
+
+The CLI design should answer the following requirements:
+
+*Core functionality*
+
+1. We should support `init`, `authn` (`login`, `authenticate`),  `policy (load)`, `list`, `show, ` `rotate_api_key` (`host` / `user`) and `fetch secrets` functionalities. These functionalities are must-haves.
+
+*Deployment*
+
+1. Provide a manual and automated way to deploy the CLI
+2. Require a maximum of 2 steps
+3. ***Recommendation:*** cross-platform deployment (Linux, Windows, macOS) should all have the same flow
```

Comment:
I think we can learn a lot from other popular CLIs here. we can review how to install [`git`](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git), [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/), [`gcloud`](https://cloud.google.com/sdk/docs/install#linux), etc

Reply:
I'd challenge this as a requirement. DAP can only run on *nix systems due to its dependency on the Kernel Keyring. As this document is specifically focused on Dockerless DAP, I don't feel cross-OS functionality is relevant.

If this document is intended to embark on a meaningful change to our CLI (which I strongly support 😄 ), I recommend we include the long term timeline of a robust, easy to use, OS native installable CLI. This will allow us to frame the shorter term deliverables required to achieve Dockerless DAP as a step toward the future desired outcome.

Reply:
This point might be confusing because it is under "requirements" but this was only a recommendation for a better UX- to have a singular flow. If I understood @InbalZilberman correctly. this was something to be explored, not necessary a "much have"